etc/init.d/zfs.in

```diff
@@ -0,0 +1,74 @@
+# ZoL userland configuration.
+
+# Run `zfs mount -a` during system start?
+ZFS_MOUNT='yes'
+
+# Run `zfs unmount -a` during system stop?
+ZFS_UNMOUNT='yes'
+
+# Run `zfs share -a` during system start?
+# nb: The shareiscsi, sharenfs, and sharesmb dataset properties.
+ZFS_SHARE='yes'
```

Comment:
> They should tell the sysadmin (via NEWS.Debian for example) that this behavior has changed.
> 
Well, if we talk about the debs I provide, that was changed a very, very long time ago. I can't even find when it was changed (because I deleted the original file I used for this file and replaced it with the one from the PR). But if I had to guess, sometimes in the 0.6.2 series.

And again, if a user have the previous default in their config, this still won't be changed. The config file is NOT updated (unless specifically told to).

So this new default is ONLY for new installs (where I still consider that doing automatic sharing is valid - for the majority of users, but granted, not for you and other users). But for the majority, it is by default [to do the automatic sharing].

And I do not consider this a change that I need to tell users about. It is in their own interest to look over any config file for the service and change it appropriately for ANY new package they install and use.
> I never stated that my situation should be applied to all users.
> 
You said:
> Sharing by default is a change in the default behavior. If you're going to do this, please make sure it's included in the news of any packages that are built.
> 
… and I have tried to explain that I really haven't _changed_ anything (for existing users).

> It makes it more difficult to have a productive conversation with you.
> 
If you can't handle the fact that others have different opinion on this than you and you try to _force_ your opinion, maybe you shouldn't comment? You have iterated the exact same arguments the whole time, without supplying me with others when I have refuted yours.

I do not agree with you, I've said that from the start.

But since this isn't really relevant for this PR, but for the debs I provide, this discussion should be moved to the pkg-zfs repository tracker.

For this PR, which is "new" (for everyone else), I consider that doing the automatic sharing (and mounting) is a reasonable default for the majority (and being "new", there's no need to document this "change" - because there IS no change!).

For those that disagree [that this should be done on their system], it is a config file and SHOULD be modified for local specifications.